Package.swift

```diff
@@ -539,10 +540,20 @@ let package = Package(
             exclude: ["Inputs/Foo.pc"]
         ),
 
+        // API intended for use by package clients.
```

Comment:
I like this as it sets the stage for a clean room implementation of a public API. I wonder if we should take it a step further and call this module "SwiftPMClient" or "libSwiftPM" or something along these lines and then over time pour into it those APIs we _want_ to expose as the stable API of SwiftPM when used as a library.

Reply:
@neonichu @abertelrud wdyt?

Reply:
I think if we are taking a step-by-step approach, renaming this to something like `libSwiftPM` should be more of the final step? Generally, I like this approach, though. I think we can extend this over time and once we *think* this could be the actual public API, we should take another pass over it for consistency etc and then we could rename it and make it official.

Reply:
This seems like a good approach, but will need to be extra careful about what goes into that module.  Would it also be a good time to add another .docc for that module so that we add documentation along side it from the start?

Reply:
At the very least each public entry point in the module should have good documentation if this is intended to grow into a cleaner API for clients to use instead of the current modules.

Reply:
I worry we get stuck with the name but not a strong opinion.